<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerCropVolumeModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerCropVolumeModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>670</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Crop Volume</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="ctkCollapsibleButton" name="ParametersCollapsibleButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Parameters</string>
     </property>
     <layout class="QFormLayout" name="parametersFormLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <property name="verticalSpacing">
       <number>12</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="ParameterNodeLabel">
        <property name="text">
         <string>Parameters Node:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="ParametersNodeComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLCropVolumeParametersNode</string>
         </stringlist>
        </property>
        <property name="showHidden">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="InputVolumeLabel">
        <property name="text">
         <string>Input volume:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="InputVolumeComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="InputROILabel">
        <property name="text">
         <string>Input ROI:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLNodeComboBox" name="InputROIComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLAnnotationROINode</string>
         </stringlist>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="editEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="ctkCheckBox" name="VisibilityButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Display ROI</string>
          </property>
          <property name="indicatorIcon">
           <iconset resource="../../../../../Libs/MRML/Widgets/Resources/qMRMLWidgets.qrc">
            <normaloff>:/Icons/VisibleOn.png</normaloff>
            <normalon>:/Icons/VisibleOff.png</normalon>:/Icons/VisibleOn.png</iconset>
          </property>
          <property name="indicatorIconSize">
           <size>
            <width>21</width>
            <height>21</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="ROIFitPushButton_2">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Fit to Volume</string>
          </property>
          <property name="icon">
           <iconset resource="../../../../../Libs/MRML/Widgets/Resources/qMRMLWidgets.qrc">
            <normaloff>:/Icons/ViewCenter.png</normaloff>:/Icons/ViewCenter.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="TechniqueLabel">
        <property name="text">
         <string>Technique:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QRadioButton" name="VoxelBasedModeRadioButton">
        <property name="text">
         <string>Voxel based cropping</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">TechniqueButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QRadioButton" name="InterpolationModeRadioButton">
        <property name="text">
         <string>Interpolated cropping</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">TechniqueButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="OutputVolumeLabel">
        <property name="text">
         <string>Output volume:</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="qMRMLNodeComboBox" name="OutputVolumeComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="baseName">
         <string>Cropped volume</string>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QFrame" name="InterpolationOptionsFrame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="IsotropicOutputVoxelLabel">
           <property name="text">
            <string>Isotropic spacing:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="IsotropicCheckbox">
           <property name="text">
            <string/>
           </property>
           <property name="tristate">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="InputSpacingScalingConstantLabel">
           <property name="toolTip">
            <string>If not equal to 1, this will result in upsampling (&lt;1) or downsampling (&gt;1) relative to the voxel spacing of the input volume.</string>
           </property>
           <property name="text">
            <string>Spacing scale:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="InterpolatorLabel">
           <property name="text">
            <string>Interpolator:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>6</number>
           </property>
           <item>
            <widget class="QRadioButton" name="NNRadioButton">
             <property name="text">
              <string>Nearest Neighbor</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="LinearRadioButton">
             <property name="text">
              <string>Linear</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="WSRadioButton">
             <property name="text">
              <string>Windowed Sinc</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="BSRadioButton">
             <property name="text">
              <string>B-spline</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="1" column="1">
          <widget class="ctkDoubleSpinBox" name="SpacingScalingSpinBox">
           <property name="toolTip">
            <string>The voxel spacing in the output volume will be scaled by this value.</string>
           </property>
           <property name="suffix">
            <string>x</string>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="OutputVolumeLabel_2">
        <property name="text">
         <string>Dimensions:</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="OutputVolumeLabel_3">
        <property name="text">
         <string>Spacing:</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="qMRMLCoordinatesWidget" name="ImageDimensionsWidget">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Output volume dimension after cropping</string>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000000.000000000000000</double>
        </property>
        <property name="coordinates" stdset="0">
         <string>1,1,1</string>
        </property>
        <property name="unitAwareProperties">
         <set>qMRMLCoordinatesWidget::All</set>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="qMRMLCoordinatesWidget" name="ImageSpacingWidget">
        <property name="toolTip">
         <string>Output volume spacing after cropping</string>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="decimalsOption">
         <set>ctkDoubleSpinBox::DecimalsByKey|ctkDoubleSpinBox::DecimalsByShortcuts|ctkDoubleSpinBox::DecimalsByValue</set>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000000000.000000000000000</double>
        </property>
        <property name="coordinates" stdset="0">
         <string>0,0,0</string>
        </property>
        <property name="quantity">
         <string>length</string>
        </property>
        <property name="unitAwareProperties">
         <set>qMRMLCoordinatesWidget::Precision|qMRMLCoordinatesWidget::Prefix|qMRMLCoordinatesWidget::Scaling|qMRMLCoordinatesWidget::Suffix</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="CropButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLCoordinatesWidget</class>
   <extends>ctkCoordinatesWidget</extends>
   <header>qMRMLCoordinatesWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCheckBox</class>
   <extends>QCheckBox</extends>
   <header>ctkCheckBox.h</header>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCoordinatesWidget</class>
   <extends>QWidget</extends>
   <header>ctkCoordinatesWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkDoubleSpinBox</class>
   <extends>QWidget</extends>
   <header>ctkDoubleSpinBox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../../Libs/MRML/Widgets/Resources/qMRMLWidgets.qrc"/>
  <include location="../qSlicerCropVolumeModule.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>qSlicerCropVolumeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ParametersNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>4</y>
    </hint>
    <hint type="destinationlabel">
     <x>376</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerCropVolumeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>InputVolumeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>4</y>
    </hint>
    <hint type="destinationlabel">
     <x>376</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerCropVolumeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>InputROIComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>7</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>InterpolationModeRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>InterpolationOptionsFrame</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>429</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerCropVolumeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>OutputVolumeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>518</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerCropVolumeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ImageDimensionsWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>615</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerCropVolumeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ImageSpacingWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>660</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="TechniqueButtonGroup"/>
 </buttongroups>
</ui>
